internal/core/drand_beacon_control.go

```diff
@@ -186,15 +185,15 @@ func (bp *BeaconProcess) Status(ctx context.Context, in *drand.StatusRequest) (*
 
 		if err == nil && lastBeacon != nil {
 			chainStore.IsEmpty = false
-			chainStore.LastRound = lastBeacon.GetRound()
-			chainStore.Length = lastBeacon.GetRound() + 1
+			chainStore.LastStored = lastBeacon.GetRound()
+			chainStore.ExpectedLast = common2.CurrentRound(bp.opts.clock.Now().Unix(), bp.group.Period, bp.group.GenesisTime)
 		}
 	}
 
 	// remote network connectivity
 	nodeList := in.GetCheckConn()
 	// in case of an empty list, we test all nodes in the group file
-	if bp.beacon != nil && bp.group != nil {
+	if len(nodeList) == 1 && bp.beacon != nil && bp.group != nil {
```

Comment:
how come if `== 1?` also we loop over `nodeList` inside the if, which doesn't make sense if its length is 1

Reply:
Ah that should have been `== 0`. 

We loop over nodes in the group to add them to `nodeList`, no?

Reply:
if `len(nodeList) == 0`, and group files won't have duplicate entries, it seems like the internal loop `for _, nl := range nodeList` is unnecessary, we can just do the `append` for every entry in `bp.group.Nodes`

Reply:
ahh, is the point that we need to check non-empty `nodeList` entries for existence in the group file? 
I guess in that case the way would be to check `len(nodeList) > 0`, and turn the group file into a `map[string]bool` for quick comparison, and it should be a separate codepath to filling an empty `nodeList`